docs(lambda-extension): update distributed tracing headers warning; (#28319)
hashicorp/vault: website/content/docs/platform/aws/lambda-extension.mdx

Currently, the cache key is a hash of the request URL path, headers, body, and
token.

<Warning title="Nonstandard distributed tracing headers may negate the cache">

>   The Vault Lambda Extension cache key includes headers from proxy requests, but excludes
>   standard distributed tracing headers `traceparent` and `tracestate`, since trace IDs are unique
>   per request. However, certain distributed tracing tools may add nonstandard tracing headers. 
>   These nonstandard tracing headers headers can lead to individualized hashes that make repeated 
>   requests unique and cause cache misses.

Reviewer: ```suggestion
  The Vault Lambda Extension cache key includes headers from proxy requests, but 
  excludes the standard distributed tracing headers `traceparent` and
  `tracestate` because trace IDs are unique per request and would lead to unique
  hashes for repeated requests.
  
  Some distributed tracing tools may add nonstandard tracing headers, which can 
  also lead to individualized hashes that make repeated requests unique and 
  cause cache misses.
```

Slight language tweaks to better tie the two statements together
